src/twisted/conch/test/test_endpoints.py

```diff
@@ -1435,31 +1435,37 @@ def test_cleanupConnectionImmediately(self):
         helper.cleanupConnection(object(), True)
 
 
+class _WriteDiscarder(BytesIO):
+    def write(self, data: object, /) -> int:
+        """
+        Discard writes because we are emulating a console object, where they'd
+        go to the screen, not back into the buffer to be read like an a+ file.
+        """
+        return 0
```

Comment:
can we assert that we write str or bytes  ?

I thins that ByteIO will reject `str` for a write operation, but it looks _WriteDiscarder will accept anything.


If this is on purpose, I think that this needs to mentioned in the docstring of `_WriteDiscarder`